app/controllers/lti_controller.rb

```diff
@@ -63,6 +68,41 @@ def init
 
     private
 
+    def populate_remote_user
+        unless @user
+            assignment = find_remote_user_assignment
+            user_by_email = find_lti_user_by_eamil
+            if assignment.present? && user_by_email.present? && assignment.should_lti_populate_remote_user?
+                # in the data base remote_user_id is username
+                assignment.set(:username => @lti_info[:login_id])
+                @user = user_by_email
+            end
+        end
+    end
+
+    def find_remote_user_assignment
+
+        assignments = UserAssignment.joins(:user).where( {
+            :user_assignments => { :organization_id => @organization.self_and_descendants}, 
+            :users => { :email => @lti_info[:email] }
+        } )
+
+        return nil unless assignments.count == 1
+        assignments.first
+    end
+
+    def find_lti_user_by_eamil
+        
+        users = User.joins(:user_assignments).where( {
+            :user_assignments => { :organization_id => @organization.self_and_descendants }, 
+            :users => { :email => @lti_info[:email] }
+        } )
+
+        return nil unless users.count == 1 
+        return nil if users.first.is_admin?
```

Comment:
should check for any global permission, if any global permission is on the user record, don't assign this way.